doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -7,14 +7,58 @@ Security
    :local:
    :depth: 3
 
-Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into four major categories:
 
+* Trusted Execution Environment
 * Cryptography
 * Trusted storage
 * Securing production devices
 
 In the following sections you will learn more details about each listed category.
 
+Trusted Execution Environment
+*****************************
+
+In Matter samples, you can use two types of Trusted Execution Environments (TEE) to provide a secure environment for the execution of security-sensitive code, depending on the SoC.
+The Trusted Firmware-M (TF-M) is dedicated to the nRF54L SoC series, while the nRF54H SoC series can use the Secure Domain.
+All cryptographic operations are performed using the PSA Cryptography API.
+
+nRF54H Secure Domian
+====================
+
+The secure domain is a dedicated domain that executes a pre-compiled, Nordic Semiconductor-signed firmware component.
+This firmware allows the execution of all secure operations in a separate domain, providing a secure environment for security-sensitive Matter code.
+
+In Matter samples, we use security services exposed by the secure domain to perform all cryptographic operations during commissioning to Matter and while operating in the Matter network.
+All keys used by Matter, such as session keys, DAC private keys, and other keys, are stored in the secure domain's secure space and are isolated from local domains.
+
+To learn more about the secure domain, see the :ref:`ug_nrf54h20_architecture_cpu` user guide.
+
+nRF54L Trusted Firmware-M (TF-M)
+================================
+
+Matter samples based on the nRF54L SoC support the Trusted Firmware-M (TF-M) library.
+All cryptographic operations within the Matter stack are performed by utilizing the PSA Cryptography API and executed in the secure TF-M environment.
+Application and Matter codes are executed in the non-secure application, whereas MCUboot and TF-M are executed in the secure part.
+Both parts are isolated from each other, providing a secure environment for the execution of security-sensitive code.
+The secure materials like Matter Session keys, DAC private key and other keys, are stored in the TF-M secure storage utilizing the :ref:`tfm_encrypted_its` module.
+Matter samples utilize the full TF-M library, so you cannot use the :ref:`tfm_minimal_build` TF-M build variant.
```

Comment:
```suggestion
Matter samples utilize the full TF-M library, so you cannot use the :ref:`tfm_minimal_build` version of TF-M.
```
Link text is "Minimal build" so this works better than repeating "build variant".